cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java

```diff
@@ -425,4 +431,28 @@ public static void loadConfigurations(Path sourceRootPath, Map<String, String> r
             throw new BLangRuntimeException(e.getMessage(), e);
         }
     }
+
+    private static int getStatusCode(BMap<String, BValue> errorDetails) {
+        if (!errorDetails.hasKey(STATUS_CODE)) {
+            return DEFAULT_ERROR_RETURN_STATUS_CODE;
+        }
+
+        BValue specifiedStatusCode = errorDetails.get(STATUS_CODE);
+        if (specifiedStatusCode.getType().getTag() != TypeTags.INT) {
```

Comment:
I am thinking whether we should special case, `int? status_code`. But checking int is sufficient for now.

Reply:
Since this is runtime validation, this is not a problem.